test/helpers/position_assertions.cc

```diff
@@ -1182,50 +1289,14 @@ void SymbolSearchAssertion::checkAllForQuery(std::string query,
             symbols.push_back(shared_ptr<SymbolInformation>(std::move(symbolInfo)));
         }
     }
-    std::map<std::string, vector<shared_ptr<SymbolInformation>>> symbolsByLine;
-    for (auto &symbol : symbols) {
-        auto &location = symbol->location;
-        auto line = fmt::format("{}:{:{}}", location->uri, location->range->start->line, lineWidth);
-        symbolsByLine[line].push_back(symbol);
-    }
-
-    bool success = true;
-    // Show any expected symbols that weren't returned by LSP
-    for (auto entry : assertionsByLine) {
-        auto lineInfo = entry.first;
-        auto expected = entry.second;
-        auto &actual = symbolsByLine[lineInfo];
-        if (expected.size() == actual.size()) {
-            continue;
-        }
-        success = false;
-        auto assertion = expected[0];
-        auto location = assertion->getLocation(uriPrefix);
-        auto sourceLine = getLine(sourceFileContents, uriPrefix, *location.get());
-        ADD_FAILURE_AT(location->uri.c_str(), assertion->range->start->line + 1)
-            << fmt::format("WorkspaceSymbol query `{}` did not return a reference to:\n{}\n", query,
-                           prettyPrintRangeComment(sourceLine, *assertion->range, ""));
-    }
-
-    // Show any symbols returned by LSP that weren't asserted
-    for (auto entry : symbolsByLine) {
-        auto lineInfo = entry.first;
-        auto actual = entry.second;
-        auto &expected = assertionsByLine[lineInfo];
-        if (expected.size() == actual.size()) {
-            continue;
-        }
-        success = false;
-        auto symbolInfo = actual[0];
-        auto &location = symbolInfo->location;
-        auto sourceLine = getLine(sourceFileContents, uriPrefix, *location.get());
-        ADD_FAILURE_AT(location->uri.c_str(), location->range->start->line + 1)
-            << fmt::format("WorkspaceSymbol query `{}` returned an unexpected reference to:\n{}\n", query,
-                           prettyPrintRangeComment(sourceLine, *location->range, ""));
-    }
 
-    ASSERT_TRUE(success) << fmt::format("For query \"{}\"", query);
+    vector<pair<shared_ptr<SymbolSearchAssertion>, shared_ptr<SymbolInformation>>> assertionSymbolPairs;
+    vector<shared_ptr<SymbolSearchAssertion>> unmatchedAssertions;
+    vector<shared_ptr<SymbolInformation>> unmatchedSymbols;
```

Comment:
Can these three be local variables in `matchAssertionsToSymbols`? Right now, you're passing them by reference to `matchAssertionsToSymbol`, and then never using them again.

Reply:
Alternatively, could you inline the contents of `matchAssertionsToSymbols` here?

Reply:
You're totally right...inlined!